packages/generator/src/utils/get-codegen.ts

```diff
@@ -0,0 +1,101 @@
+import {CodegenConfig, NextConfigComplete} from "next/dist/server/config-shared"
+import {baseLogger} from "next/dist/server/lib/logging"
+
+export const defaultCodegenConfig: CodegenConfig = {
+  fieldTypeMap: {
+    string: {
+      component: "LabeledTextField",
+      inputType: "text",
+      zodType: "string",
+      prismaType: "String",
+    },
+    boolean: {
+      component: "LabeledTextField",
+      inputType: "text",
+      zodType: "boolean",
+      prismaType: "Boolean",
+    },
+    int: {
+      component: "LabeledTextField",
+      inputType: "number",
+      zodType: "number",
+      prismaType: "Int",
+    },
+    number: {
+      component: "LabeledTextField",
+      inputType: "number",
+      zodType: "number",
+      prismaType: "Int",
+    },
+    bigint: {
+      component: "LabeledTextField",
+      inputType: "number",
+      zodType: "number",
+      prismaType: "BigInt",
+    },
+    float: {
+      component: "LabeledTextField",
+      inputType: "number",
+      zodType: "number",
+      prismaType: "Float",
+    },
+    decimal: {
+      component: "LabeledTextField",
+      inputType: "number",
+      zodType: "number",
+      prismaType: "Decimal",
+    },
+    datetime: {
+      component: "LabeledTextField",
+      inputType: "text",
+      zodType: "string",
+      prismaType: "DateTime",
+    },
+    uuid: {
+      component: "LabeledTextField",
+      inputType: "text",
+      zodType: "string().uuid",
+      prismaType: "String",
+      default: "uuid",
+    },
+    json: {
+      component: "LabeledTextField",
+      inputType: "text",
+      zodType: "any",
+      prismaType: "Json",
+    },
+  },
+}
+
+export const getResourceValueFromCodegen = async (
+  fieldType: keyof CodegenConfig["fieldTypeMap"],
+  resource: keyof CodegenConfig["fieldTypeMap"][string],
+): Promise<string | undefined> => {
+  const codegen = (await getCodegen()).codegen as CodegenConfig
+  const templateValue = codegen.fieldTypeMap[fieldType][resource]
+  return templateValue
+}
+
+export const getResourceConfigFromCodegen = async (
+  fieldType: keyof CodegenConfig["fieldTypeMap"],
+): Promise<CodegenConfig["fieldTypeMap"][string]> => {
+  const codegen = (await getCodegen()).codegen as CodegenConfig
+  const config = codegen.fieldTypeMap[fieldType] || {}
+  return config
+}
+
+export const getCodegen = async (): Promise<Pick<NextConfigComplete, "codegen">> => {
```

Comment:
We could simplify it to return `codegen` directly instead of an `{codegen: codegen}` object